### ISSUE TITLE
Gen 3 Pokeball Issue

### ISSUE THREAD

**Issue by HaxMaster5235** [kwsch/PKHeX#1432]
In the most recent PkHex build I could find, from PokEdit, Pokemon from Gen 3 games are often flagged with the error, "Invalid: Can't have ball for encounter type."

I've personally Pal Parked these Pokemon from gen 3 to gen 4 using Desmume, and yet PkHex wrongly flags them unless the ball is a Pokeball. Now, I know for a fact that you can have a Butterfree in a Great Ball, and a Nidoking caught in Fire Red's Safari Zone will always be in a safari ball. Please fix this?

Also, I find the need to use RNG Reporter to generate valid PID's kind of annoying, it's not too bad, but can someone implement a PID generator modeled on the RNG Reporter one in PkHex, like PokeGen had, but actually useful?

**Comment by HaxMaster5235** [comment]
Screenshot for proof. http://prntscr.com/gdx91p

**Comment by javierhimura** [comment]
Could you upload the pkm file to check why is flagged illegal?

**Comment by HaxMaster5235** [comment]
Would you rather I upload my entire sav file, since it plagues most of my gen 3 pokemon?

**Comment by HaxMaster5235** [comment]
[NIDOKING.zip](https://github.com/kwsch/PKHeX/files/1255859/NIDOKING.zip)
Here

**Comment by HaxMaster5235** [comment]
[POKEMON PL.0.zip](https://github.com/kwsch/PKHeX/files/1255863/POKEMON.PL.0.zip)
And here's the save. I'm working on figuring out RNG reporter to fix PID's, but that seems to only be the second most common error...

**Comment by HaxMaster5235** [comment]
javierhimura? Hello...?

**Comment by kwsch** [comment]
Doesn't have a PIDIV combo, thus not a wild capture (and must be an egg). Hold control when requesting the analysis results for a full parse.

**Comment by HaxMaster5235** [comment]
Um... Ok? For the first thing, I think PkHex accurately summed that up. I was asking if something could be done about it, not what it is. Error messages seem to be very straight forward, RNG Reporter, on the other hand, not so much.

On holding Control... I'm not entirely clear on what I'm doing? Like, what am I supposed to do and then hold control, and why? 

As for downloading from external sites, I usually get it from Project Pokemon, as do probably a ton of people. I only got from Pokedit because it was a newer version, and I wanted to see if this Pokeball bug, *which is the main issue*, was fixed, but no dice. 

Now, can we focus on the issue at hand, the very NAME of this issue? When Pokemon are Pal Parked from gen 3 to gen 4, they keep their original balls, the ones they were caught in. PkHex says otherwise, says it is invalid unless it's in a normal Pokeball. 

Now, someone needs to fix this. I've included ONE of the offending pokemon, AND the save file itself, so people can see that this IS able to be recreated easily, someone just needs to fix it.
